src/h3lib/lib/polyfill.c

```diff
@@ -648,4 +679,68 @@ H3Error H3_EXPORT(polygonToCellsExperimental)(const GeoPolygon *polygon,
         out[i++] = iter.cell;
     }
     return iter.error;
+}
+
+static int MAX_SIZE_CELL_THRESHOLD = 10;
+
+static double getAverageCellArea(int res) {
+    double hexAreaKm2;
+    H3_EXPORT(getHexagonAreaAvgKm2)(res, &hexAreaKm2);
+    return hexAreaKm2;
+}
+
+/**
+ * maxPolygonToCellsSize returns the number of cells to allocate space for
+ * when performing a polygonToCells on the given GeoJSON-like data structure.
+ * @param geoPolygon A GeoJSON-like data structure indicating the poly to fill
+ * @param res Hexagon resolution (0-15)
+ * @param out number of cells to allocate for
+ * @return 0 (E_SUCCESS) on success.
+ */
+H3Error H3_EXPORT(maxPolygonToCellsSizeExperimental)(const GeoPolygon *polygon,
+                                                     int res, uint32_t flags,
+                                                     int64_t *out) {
+    // Initialize the iterator without stepping, so we can adjust the res and
+    // flags before we start
+    IterCellsPolygonCompact iter = _iterInitPolygonCompact(polygon, res, flags);
+
+    if (iter.error) {
+        return iter.error;
+    }
+
+    // Ignore the requested flags and use the faster overlapping-bbox mode
+    iter._flags = CONTAINMENT_OVERLAPPING_BBOX;
```

Comment:
Why accept a flags variable here and initialize with it if you overwrite it immediately afterwards?